src/Pos/Block/Pure.hs

```diff
@@ -584,4 +625,4 @@ verifyBlocks curSlotId verifyNoUnknown bvd initLeaders = view _3 . foldl' step s
                 , vbpMaxSize = Just (bvdMaxBlockSize bvd)
                 , vbpVerifyNoUnknown = verifyNoUnknown
                 }
-        in (newLeaders, Just $ getBlockHeader blk, res <> verifyBlock vbp blk)
+        in (newLeaders, newPsks, Just $ getBlockHeader blk, res <> verifyBlock vbp blk)
```

Comment:
To be fair, I tried hard to understand this logic but I don't understand it.

What does this `ProxySKHeavyMap` inside `VerifyBlocksIter` mean? I see that initially it's constructed from data from headers. Then you use `pskHeavyMapApplyBlock` which puts data from body there. Next thing I don't understand is why you collect heavy certs by taking them from bodies of blocks which are about to be verified? You should also consider older certs, shouldn't you?